In [2]:
import os

import requests
import xmltodict
from dotenv import load_dotenv

load_dotenv()

SERVICE_KEY = os.getenv("DATA_GO_KR_SERVICE_KEY")
query_param = {
    "serviceKey": SERVICE_KEY,
    "strtYymm": "201601",
    "endYymm": "201601",
    # "hsSgn": "1001999090",
    "cntyCd": "US",
}
response = requests.get("http://apis.data.go.kr/1220000/nitemtrade/getNitemtradeList", params=query_param)
response.raise_for_status()

data = xmltodict.parse(response.content)

In [6]:
data

{'response': {'header': {'resultCode': '00', 'resultMsg': '정상서비스.'},
  'body': {'items': {'item': [{'balPayments': '1709825452',
      'expDlr': '5027618286',
      'expWgt': '1241496326',
      'hsCd': '-',
      'impDlr': '3317792834',
      'impWgt': '1509874932',
      'statCd': '-',
      'statCdCntnKor1': '-',
      'statKor': '-',
      'year': '총계'},
     {'balPayments': '-579245',
      'expDlr': '0',
      'expWgt': '0',
      'hsCd': '0101211000',
      'impDlr': '579245',
      'impWgt': '5000',
      'statCd': 'US',
      'statCdCntnKor1': '미국',
      'statKor': '농가 사육용',
      'year': '2016.01'},
     {'balPayments': '-519540',
      'expDlr': '0',
      'expWgt': '0',
      'hsCd': '0101219000',
      'impDlr': '519540',
      'impWgt': '3050',
      'statCd': 'US',
      'statCdCntnKor1': '미국',
      'statKor': '기타',
      'year': '2016.01'},
     {'balPayments': '-1263787',
      'expDlr': '0',
      'expWgt': '0',
      'hsCd': '0105111000',
      'impDlr': '1263787',

In [8]:
import pandas as pd

pd.set_option('display.max_rows', 10)
dataset = pd.DataFrame(data["response"]["body"]["items"]["item"])
dataset

balPayments      expDlr      expWgt        hsCd      impDlr      impWgt  \
0     1709825452  5027618286  1241496326           -  3317792834  1509874932   
1        -579245           0           0  0101211000      579245        5000   
2        -519540           0           0  0101219000      519540        3050   
3       -1263787           0           0  0105111000     1263787        1431   
4          18072       21179         889  0106191000        3107         116   
...          ...         ...         ...         ...         ...         ...   
6401       -7231           0           0  9703002000        7231        1979   
6402       -1512           0           0  9704001000        1512           2   
6403      -37615           0           0  9705000000       37615         145   
6404      -40304           0           0  9706009000       40304         155   
6405        -163           0           0  9999999290         163      388633   

     statCd statCdCntnKor1                                            statKor  \
0         -              -                                                  -   
1        US             미국                                             농가 사육용   
2        US             미국                                                 기타   
3        US             미국                                                번식용   
4        US             미국                                                  개   
...     ...            ...                                                ...   
6401     US             미국                                             조상(彫像)   
6402     US             미국                                                 우표   
6403     US             미국  수집품과 표본[동물학ㆍ식물학ㆍ광물학ㆍ해부학ㆍ사학ㆍ고고학ㆍ고생물학ㆍ민족학ㆍ고전학(古錢...   
6404     US             미국                                                 기타   
6405     US             미국                                                  -   

         year  
0          총계  
1     2016.01  
2     2016.01  
3     2016.01  
4     2016.01  
...       ...  
6401  2016.01  
6402  2016.01  
6403  2016.01  
6404  2016.01  
6405  2016.01  

[6406 rows x 10 columns]

## 무역 데이터 API 응답 필드

| 필드       | 설명                        | 예시                  | 상세 설명       |
|-------------|---------------------------|----------------------|-------------|
| year       | 연도                        | 2016.01             | 기간 정보       |
| statCdCntnKor1 | 국가명                      | 미국                 | 국가명 정보      |
| statCd     | 국가코드                    | US                   | ISO 국가 코드   |
| statKor    | 품목명                      | 쇠고기(냉동한 것으로 한정) | 품목명 정보      |
| hsCd       | HS코드                      | 0202                 | 국제 무역 품목 코드 |
| expWgt     | 수출중량 (kg)               | 18,003               | 수출된 상품의 중량  |
| expDlr     | 수출금액 (달러)             | 150,822              | 수출된 상품의 금액  |
| impWgt     | 수입중량 (kg)               | 14,809,834           | 수입된 상품의 중량  |
| impDlr     | 수입금액 (달러)             | 93,754,113           | 수입된 상품의 금액  |
| balPayments | 무역수지 (달러)             | -93,603,291          | 수출금액 - 수입금액 |